### PR TITLE
Mobile Design: Staff Pets index table should render in cards below width xl: 1200px

### PR DESCRIPTION
**Pull request by sarvaiyanidhi**
# 🔗 Issue
#694 

# ✍️ Description
For Staff Pets Index view, we will be displaying table layout if width is above 1200px else card layout for smaller screen

# 📷 Screenshots/Demos

Large Screen

Alta-Staff-Pets-table-

Small Screen

Alta-Staff-Pets-Card

### CODE REVIEW

File: app/views/organizations/staff/pets/_pet_table.html.erb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<div class="card">
+  <!-- table -->
+  <div class="table-responsive overflow-y-hidden">
+    <table class="table mb-0 text-nowrap table-hover table-centered">
+      <thead>
+        <tr>
```
Comment (jmilljr24) [inline app/views/organizations/staff/pets/_pet_table.html.erb:6]: This `<tr>` had `style="height: 70px;"` previously in the index file. It was there to fix a small issue when only one pet was displayed in the table, the settings dropdown for that pet was cut off due to the responsive table overflow being hidden. We should be able to remove the responsive class from the table and avoid using the 70px height now that we will have a breakpoint at 1200px.
```suggestion
  <div>
    <table class="table mb-0 text-nowrap table-hover table-centered">
      <thead>
        <tr>
```

File: app/views/organizations/staff/pets/_pet_cards.html.erb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+      <%= image %>
+      <ul class="list-group list-group-flush">
+        <li class="list-group-item d-flex justify-content-between">
+          <h5 class="card-title text-secondary mb-0">
+            <%= pet.name %>
+          </h5>
+        </li>
+        <% li_classes = %w[list-group-item text-secondary] %>
+        <%= tag.li("Sex: #{pet.sex}", class: li_classes) %>
```
Comment (jmilljr24) [inline app/views/organizations/staff/pets/_pet_cards.html.erb:15]: @kasugaijin Do we need the breed in this card? The issue lists it but the current table does not have breed.

**Review by jmilljr24 — CHANGES_REQUESTED** [review]
@sarvaiyanidhi Thank you for working on this!! A couple of small changes that should be simple enough.

Two separate things to note if you are up for working on, if not we can open up a new issue.
1. The toggles no longer work as they should in the cards
2. It would be nice to have the cards wrap with flex instead of just a single vertical line
Screenshot from 2024-05-15 15-46-27_edit

**Comment by sarvaiyanidhi** [comment]
Thank you for the feedback @jmilljr24 .. Have noted all points and will push fixes soon to close on this.

**Comment by jmilljr24** [comment]
> Thank you for the feedback @jmilljr24 .. Have noted all points and will push fixes soon to close on this.

I appreciate it! I was looking around a little more and figured out the toggles. The link to the pet page encompasses the entire card. Maybe just have that link on the pet name or name and picture (whatever works best).

**Comment by sarvaiyanidhi** [comment]
> > Thank you for the feedback @jmilljr24 .. Have noted all points and will push fixes soon to close on this.
> 
> I appreciate it! I was looking around a little more and figured out the toggles. The link to the pet page encompasses the entire card. Maybe just have that link on the pet name or name and picture (whatever works best).

Yeah I too figured it out after you pointed out the issue. Will fix it

File: app/views/organizations/staff/pets/_pet_cards.html.erb (hunk 2 above)
Comment (kasugaijin) [reply app/views/organizations/staff/pets/_pet_cards.html.erb:15]: @jmilljr24 @sarvaiyanidhi we removed breed from the table in a PR today. So can remove from the cards also :)

**Comment by kasugaijin** [comment]
@all-contributors
please add @sarvaiyanidhi for code.

File: app/assets/stylesheets/custom.scss [hunk 3]
```diff
@@ -6,6 +6,19 @@
...
+    display: none; /* Hide the table layout on smaller screens */
+  }
+}
+
+@media (min-width: 1200px) {
+  .pet-list-cards {
+    display: none; /* Hide the card layout on larger screens */
+  }
+}
+
```
[the remark this answers is not in the document]
Comment (sarvaiyanidhi) [reply app/assets/stylesheets/custom.scss]: Removed custom CSS and fixed cards

Alta-Staff-Pets-Card

File: app/views/organizations/staff/pets/_pet_cards.html.erb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<div class="col-lg-4 col-md-6 p-0">
+  <%= link_to adoptable_pet_path(pet), class: 'text-decoration-none' do %>
+    <div class="card card-hover m-4" style='overflow: hidden'>
+      <% image = image_tag('coming_soon.jpg', class: 'card-img-top pet-card') %>
+      <% image = image_tag pet.images.first, class: 'card-img-top pet-card' if pet.images.attached? %>
+
+      <%= image %>
```
[the remark this answers is not in the document]
Comment (sarvaiyanidhi) [reply app/views/organizations/staff/pets/_pet_cards.html.erb]: Fixed code as per suggestion

File: app/views/organizations/staff/pets/index.html.erb [hunk 5]
```diff
@@ -38,73 +38,14 @@
...
+      <div class="col-xl">
+        <!-- Table layout -->
+        <div class="pet-list-table">
+          <%= render partial: "pet_table", locals: { pets: @pets} %>
+        </div>          
+        <!-- Card-based layout -->
+        <div class="pet-list-cards">
+      <%= render partial:"pet_cards", collection: @pets, as: :pet %>
         </div>
       </div>
```
[the remark this answers is not in the document]
Comment (sarvaiyanidhi) [reply app/views/organizations/staff/pets/index.html.erb]: Updated suggested code

File: app/views/organizations/staff/pets/_pet_table.html.erb (hunk 1 above)
Comment (sarvaiyanidhi) [reply app/views/organizations/staff/pets/_pet_table.html.erb]: Removed responsive table overflow being hidden

**Comment by sarvaiyanidhi** [comment]
@kasugaijin @jmilljr24 Have addressed all points and pushed updated code. Please let me know if you need any other changes. Thanks

**Review by jmilljr24 — CHANGES_REQUESTED** [review]
@sarvaiyanidhi The pet cards look and work great!

It seems we lost the [card content container](https://getbootstrap.com/docs/5.3/components/card/#about) around the pet table. If you could add that back in then we should be all set with this.
`<div class="card">`

Missing
Screenshot from 2024-05-16 20-52-58

<br>

Correct
Screenshot from 2024-05-16 21-08-03

**Comment by sarvaiyanidhi** [comment]
> @sarvaiyanidhi The pet cards look and work great!
> 
> It seems we lost the [card content container](https://getbootstrap.com/docs/5.3/components/card/#about) around the pet table. If you could add that back in then we should be all set with this. `<div class="card">`

Sorry my mistake..Added back.

**Review by jmilljr24 — APPROVED** [review]
lgtm! Thanks @sarvaiyanidhi

**Pull request merged**